services/datamanager/builtin/builtin.go

```diff
@@ -449,6 +451,11 @@ func (svc *builtIn) Reconfigure(
 	if svc.fileDeletionBackgroundWorkers != nil {
 		svc.fileDeletionBackgroundWorkers.Wait()
 	}
+	if svcConfig.DeleteEveryNthWhenDiskFull != 0 {
```

Comment:
If we're always canceling the previous file deletion routine on a reconfigure (which you verified last time is only called if the data service or one of its collectors is changed), then we don't actually need to store `deleteEveryNth` at the service level — can just have it be a local variable that's used below in `pollFilesystem`.

There's the question of whether we actually want to stop the file deletion goroutine every time we reconfigure, rather than only if `deleteEveryNth` is changed, but that's a bigger fundamental change to the original PR. I'd just want to have one more test where you capture/sync/delete and constantly reconfigure, e.g. by changing a collector's hz by 1 every few seconds.

Reply:
I can add a follow up to change reconfigure to only cancel the file polling goroutine if deleteEveryNth or captureDsiabled is changed. I don't think itll be ready by rdk release today. I can do that test for you, do you mean manually or a unit test/what are you hoping to verify in testing that. Will also remove deleteNth off of the service.

Reply:
Yep I mean a manual test to verify that things don't go wonky, but a code change isn't necessary unless that bricks the system or Reconfigure! SGTM about just removing deleteNth off the service within this PR, then LGTM to merge

Reply:
Ok tested, no wonky behaviour was noticed. There were no spurious reconfigures/odd behaviour I saw.